File: Mobile App/Apollo Deliver/lib/Models/Shipper.dart
import 'package:apollodeliver/Models/Address.dart';
import 'package:apollodeliver/Models/Review.dart';

class Shipper {
  int? id;
  String? shipperName;
  String? password;
  String? phoneNumber;
  bool? enabled;
  Set<Address>? address;
  String? email;
  String? role;
  List<Review>? reviewList;

  Shipper({
    this.id,
    this.shipperName,
    this.password,
    this.phoneNumber,
    this.enabled,
    this.address,
    this.email,
    this.role,
    this.reviewList,
  });

  factory Shipper.fromJson(Map<String, dynamic> json) {
    return Shipper(
      id: json['id'],
      shipperName: json['shipperName'],
      password: json['password'],
      phoneNumber: json['phoneNumber'],
      enabled: json['enabled'],
      address: (json['address'] as List<dynamic>?)
          ?.map((i) => Address.fromJson(i as Map<String, dynamic>))
          .toSet(),
      email: json['email'],
      role: json['role'],
      reviewList: (json['reviewList'] as List<dynamic>?)
          ?.map((i) => Review.fromJson(i as Map<String, dynamic>))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'shipperName': shipperName,
    'password': password,
    'phoneNumber': phoneNumber,
    'enabled': enabled,
    'address': address?.map((a) => a.toJson()).toList(),
    'email': email,
    'role': role,
    'reviewList': reviewList?.map((r) => r.toJson()).toList(),
  };
}

File: API/apollo_api_services/src/main/java/com/apollo/entity/Shipper.java
package com.apollo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "shippers")
public class Shipper {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "shipper_name")
    private String shipperName;

    @Column(name = "password")
    @JsonIgnore
    private String password;

    @Column(name = "phone_number")
    private String phoneNumber;

    @Column(name = "enabled")
    private boolean enabled;

    @OneToMany(mappedBy = "shipper")
    @JsonIgnore
    private Set<Address> address;

    @Column(name = "email")
    private String email;

    @Column(name = "role")
    private String role;

    @OneToMany(mappedBy = "shipper")
    @JsonIgnore
    private List<Review> reviewList;
}
